apps/static-site/src/components/iiif/RangeNavigation.tsx

```diff
@@ -0,0 +1,45 @@
+import { useSimpleViewer, useVault, useManifest } from "react-iiif-vault";
+import type { Manifest } from "@iiif/presentation-3";
+import {
+  getValue,
+  rangesToTableOfContentsTree,
+  type RangeTableOfContentsNode,
+} from "@iiif/helpers";
+
+type RangeNavigationProps = {
+  manifest: Manifest;
+};
+export function RangeNavigation({ manifest }: RangeNavigationProps) {
+  const context = useSimpleViewer();
+  const { setCurrentCanvasId } = context;
+  const vault = useVault();
+  const mani = useManifest();
+  const structures = mani?.structures ?? [];
+  const toc = rangesToTableOfContentsTree(vault, structures);
+
+  return (
+    <div className="overflow-hidden font-mono">
+      <div className="cut-corners w-full place-self-start bg-black p-5 text-white">
+        <h3 className="mb-4 uppercase">Table of Contents</h3>
+        {toc?.items?.map((range: RangeTableOfContentsNode) => {
```

Comment:
I think if there is no table of contents, or an empty range we should return early with a `null`. Most items won't have a range I don't think.